aiida_lammps/workflows/base.py

```diff
@@ -0,0 +1,274 @@
+"""Workchain to run a LAMMPS calculation with automated error handling and restarts."""
+import re
+
+from aiida import orm
+from aiida.common import AttributeDict, InputValidationError
+from aiida.engine import (
+    BaseRestartWorkChain,
+    ProcessHandlerReport,
+    process_handler,
+    while_,
+)
+import numpy as np
+
+from aiida_lammps.calculations.base import LammpsBaseCalculation
+from aiida_lammps.utils import RestartTypes
+
+
+class LammpsBaseWorkChain(BaseRestartWorkChain):
+    """Base workchain for calculations using LAMMPS"""
+
+    _process_class = LammpsBaseCalculation
+
+    @classmethod
+    def define(cls, spec):
+        """Define the process specification."""
+        # yapf: disable
+        super().define(spec)
+        spec.expose_inputs(LammpsBaseCalculation)
+        spec.input(
```

Comment:
Why are you redefining this input? It is already defined on the base class. Same goes for the `clean_workdir`.

Reply:
Good point, I thought I still needed to make them accessible at his level, at least that is how it was made in the aiida-vasp But maybe it is because that restart workchain was written before the `BaseRestartWorkChain` was officially out.

Reply:
The `aiida-vasp` has indeed been around for a while and it may be doing some things that are no longer optimal. Frankly, for a recent example of a `BaseRestartWorkChain` implementation, I would recommend having a look at this one: https://github.com/microsoft/aiida-pyscf/blob/main/src/aiida_pyscf/workflows/base.py

That should follow all latest best practices and should provide a good example I think.